lib/renderer/web-view/web-view.js

```diff
@@ -277,69 +273,74 @@ const registerWebViewElement = function () {
+    }
+    return internal.guestInstanceId
+  }
+
   // Forward proto.foo* method calls to WebViewImpl.foo*.
-  const createBlockHandler = function (m) {
+  const createBlockHandler = function (method) {
     return function (...args) {
-      const internal = v8Util.getHiddenValue(this, 'internal')
-      if (internal.webContents) {
-        return internal.webContents[m](...args)
+      const [error, result] = ipcRenderer.sendSync('ELECTRON_BROWSER_SYNC_CALL_TO_GUEST_VIEW', getGuestInstanceId(this), method, args)
+      if (error) {
+        throw errorUtils.deserialize(error)
       } else {
-        throw new Error(`Cannot call ${m} because the webContents is unavailable. The WebView must be attached to the DOM and the dom-ready event emitted before this method can be called.`)
+        return result
       }
     }
   }
   for (const method of methods) {
     proto[method] = createBlockHandler(method)
   }
 
-  const createNonBlockHandler = function (m) {
+  const createNonBlockHandler = function (method) {
     return function (...args) {
-      const internal = v8Util.getHiddenValue(this, 'internal')
-      ipcRenderer.send('ELECTRON_BROWSER_ASYNC_CALL_TO_GUEST_VIEW', contextId, null, internal.guestInstanceId, m, ...args)
+      const callback = (typeof args[args.length - 1] === 'function') ? args.pop() : null
+      const requestId = getNextId()
+      ipcRenderer.once(`ELECTRON_RENDERER_ASYNC_CALL_TO_GUEST_VIEW_RESPONSE_${requestId}`, function (event, error, result) {
+        if (error == null) {
+          if (callback) callback(result)
+        } else {
+          throw errorUtils.deserialize(error)
+        }
+      })
+      ipcRenderer.send('ELECTRON_BROWSER_ASYNC_CALL_TO_GUEST_VIEW', requestId, getGuestInstanceId(this), method, args, callback != null)
     }
   }
   for (const method of nonblockMethods) {
     proto[method] = createNonBlockHandler(method)
   }
 
-  proto.executeJavaScript = function (code, hasUserGesture, callback) {
-    const internal = v8Util.getHiddenValue(this, 'internal')
-    if (typeof hasUserGesture === 'function') {
-      callback = hasUserGesture
-      hasUserGesture = false
-    }
-    const requestId = getNextId()
-    ipcRenderer.send('ELECTRON_BROWSER_ASYNC_CALL_TO_GUEST_VIEW', contextId, requestId, internal.guestInstanceId, 'executeJavaScript', code, hasUserGesture)
-    ipcRenderer.once(`ELECTRON_RENDERER_ASYNC_CALL_TO_GUEST_VIEW_RESPONSE_${requestId}`, function (event, result) {
-      if (callback) callback(result)
-    })
-  }
-
   // WebContents associated with this webview.
   proto.getWebContents = function () {
     const internal = v8Util.getHiddenValue(this, 'internal')
-    if (!internal.webContents) {
+    if (!internal.guestInstanceId) {
       internal.createGuestSync()
     }
-    return internal.webContents
+    return remote.getGuestWebContents(internal.guestInstanceId)
```

Comment:
Is this intended to be replaced with an appropriate ipc call in future, as per PR description I believe it was intended to remove all uses of remote in <webview> implementation.

Reply:
@deepak1556 this is the only method that still needs to use `remote` the preserve the semantics